flexget/plugins/clients/aria2.py

```diff
@@ -83,36 +87,76 @@ def on_task_output(self, task, config):
         aria2 = self.aria2_connection(config['server'], config['port'],
                                       config['username'], config['password'])
         for entry in task.accepted:
-            if task.options.test:
-                log.verbose('Would add `%s` to aria2.', entry['title'])
-                continue
-            try:
-                self.add_entry(aria2, entry, config)
-            except socket_error as se:
-                entry.fail('Unable to reach Aria2: %s', se)
-            except xmlrpc.client.Fault as err:
-                log.critical('Fault code %s message %s', err.faultCode, err.faultString)
-                entry.fail('Aria2 communication Fault')
-            except Exception as e:
-                log.debug('Exception type %s', type(e), exc_info=True)
-                raise
+            # check for content_files first, then use url or title if not
+            if 'content_files' not in entry:
+                if entry['url']:
+                    entry['content_files'] = [entry['url']]
+                else:
+                    entry['content_files'] = [entry['title']]
+            else:
+                if not isinstance(entry['content_files'], list):
+                    entry['content_files'] = [entry['content_files']]
+            for current_file in entry['content_files']:
+                # expose current filename for rendering purposes
+                entry['current_file'] = current_file
+                if task.options.test:
+                    log.verbose('Would add `%s` to aria2.', entry['current_file'])
+                    continue
+                try:
+                    self.add_entry(aria2, entry, config)
+                except socket_error as se:
+                    entry.fail('Unable to reach Aria2: %s', se)
+                except xmlrpc.client.Fault as err:
+                    log.critical('Fault code %s message %s', err.faultCode, err.faultString)
+                    entry.fail('Aria2 communication Fault')
+                except Exception as e:
```

Comment:
I think there's no need for that since task catches all Unhandled exceptions anyway